internal/plugin/host/plugin_client.go

```diff
@@ -0,0 +1,72 @@
+package host
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/plugin/builtin/testing/testhostplugin"
+	"github.com/hashicorp/boundary/sdk/pbs/plugin"
+)
+
+// embeddedPluginClients contains a factory to our embedded plugins.
+// These plugins are loaded directly and linked in the Boundary
+// binary, instead of being installed from the embedded filesystem.
+var embeddedPluginClients = map[string]func() plugin.HostPluginServiceClient{
+	// "aws": awsplugin.NewClient,
+	"testing": testhostplugin.NewClient,
+}
+
+// PluginManager is a helper for loading and managing host plugins.
+type PluginManager struct {
+	repo *Repository
```

Comment:
We probably don't want to store the repo here since I think one of the main use cases for the plugin manager is for the repo in `internal/host/plugin` to be able to use it and nested transactions are problematic and might be hard to track down when doing repo -> repo calls.

Since initially we are going to be only running a very small number of plugins (3 - 5) consider having `NewPluginmanager` take a db object instead and load the db data into memory.  That could also be the time that the plugin processes are started and the clients are created.  Then when the users of the plugin manager requests a plugin (by id, or maybe by name) the manager can return the plugin data and the plugin client w/o having to hit the db or start any processes.